VideoLocker/src/main/java/org/edx/mobile/view/custom/ECheckedTextView.java

```diff
@@ -0,0 +1,40 @@
+package org.edx.mobile.view.custom;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Typeface;
+import android.util.AttributeSet;
+import android.widget.CheckedTextView;
+
+import org.edx.mobile.R;
+
+public class ECheckedTextView extends CheckedTextView {
+    public ECheckedTextView(Context context) {
+        super(context);
+    }
+
+    public ECheckedTextView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        processAttrs(context, attrs);
+    }
+
+    public ECheckedTextView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        processAttrs(context, attrs);
+    }
+
+    private void processAttrs(Context context, AttributeSet attrs) {
+        if (isInEditMode()) return;
+        TypedArray a = context.getTheme().obtainStyledAttributes(
+                attrs, R.styleable.custom_view, 0, 0);
+        // Check for the font attribute and setup font
+        String fontFileName = a.getString(R.styleable.custom_view_font);
+        a.recycle();
+        if (fontFileName == null) {
+            // If font is not defined, then set default font
+            fontFileName = "OpenSans-Regular.ttf";
```

Comment:
There’s a solution provided for API 21 as well in one of the answers, but yeah using reflection means that you’re not relying on any documented API, and thus is subject to breakage by any provider who diverges from the AOSP.
> On 16-Oct-2015, at 10:11 PM, Brian Guertin <notifications@github.com> wrote:
> 
> In VideoLocker/src/main/java/org/edx/mobile/view/custom/ECheckedTextView.java <https://github.com/edx/edx-app-android/pull/386#discussion_r42265916>:
> 
> > +
> > +    public ECheckedTextView(Context context, AttributeSet attrs, int defStyleAttr) {
> > +        super(context, attrs, defStyleAttr);
> > +        processAttrs(context, attrs);
> > +    }
> > +
> > +    private void processAttrs(Context context, AttributeSet attrs) {
> > +        if (isInEditMode()) return;
> > +        TypedArray a = context.getTheme().obtainStyledAttributes(
> > +                attrs, R.styleable.custom_view, 0, 0);
> > +        // Check for the font attribute and setup font
> > +        String fontFileName = a.getString(R.styleable.custom_view_font);
> > +        a.recycle();
> > +        if (fontFileName == null) {
> > +            // If font is not defined, then set default font
> > +            fontFileName = "OpenSans-Regular.ttf";
> Yeah. Also that reflection solution doesn't work with API 21.
> 
> Anyway I filed a story for later: https://openedx.atlassian.net/browse/MA-1487 <https://openedx.atlassian.net/browse/MA-1487>
> —
> Reply to this email directly or view it on GitHub <https://github.com/edx/edx-app-android/pull/386/files#r42265916>.
>